Zend/zend_alloc.c

```diff
@@ -797,7 +799,10 @@ static int zend_mm_chunk_extend(zend_mm_heap *heap, void *addr, size_t old_size,
 		}
 	}
 #endif
-#ifndef _WIN32
+#if HAVE_MREMAP
+	/* We don't use MREMAP_MAYMOVE due to alignment requirements. */
+	return mremap(addr, old_size, new_size, 0) != MAP_FAILED;
```

Comment:
Little detail NetBSD has similar API only it has an additional argument (new address) and some specific flags (but should not be needed here). Do not know for you if it s better to allow it only on Linux or not.